ruleset/regexp.go

```diff
@@ -39,17 +39,47 @@ func NewRegexp(include, exclude []*regexp.Regexp) (*Regexp, error) {
 		return nil
 	}
 
-	return &Regexp{
+	return &RegexpMatcher{
 		include: build(include),
 		exclude: build(exclude),
 	}, nil
 }
 
 // Match returns true if the given string matches at least one of the include rules
 // and does not match the exclude rules.
-func (r *Regexp) Match(s string) bool {
+func (r *RegexpMatcher) Match(s string) bool {
 	if r.exclude != nil && r.exclude.MatchString(s) {
 		return false
 	}
 	return r.include != nil && r.include.MatchString(s)
 }
+
+type RegexpListItem struct {
+	*regexp.Regexp
+	exclude bool
+}
+
+func ParseRegexpListItem(v string) (RegexpListItem, error) {
+	var exclude bool
+	if v[0] == '-' {
```

Comment:
It would panic if string is empty. We should check it.

Reply:
true.